frontend/src/components/common/PrintCSVButton.tsx

```diff
@@ -0,0 +1,146 @@
+import React, { useEffect, useState } from "react";
+import {
+  Icon,
+  IconButton,
+  Box,
+  Modal,
+  ModalOverlay,
+  ModalContent,
+  ModalHeader,
+  ModalBody,
+  FormControl,
+  Button,
+  FormHelperText,
+  Flex,
+  ScaleFade,
+  Alert,
+  AlertDescription,
+  AlertIcon,
+} from "@chakra-ui/react";
+import { AiFillPrinter } from "react-icons/ai";
+import { RangeDatepicker } from "chakra-dayzed-datepicker";
+import commonAPIClient from "../../APIClients/CommonAPIClient";
+import CSVConverter from "../../helper/CSVConverter";
+
+const PrintCSVButton = (): React.ReactElement => {
+  const [selectedDates, setSelectedDates] = useState<Date[]>([]);
+  const [showAlert, setShowAlert] = useState(false);
+
+  const handleClear = () => {
+    setSelectedDates([]);
+  };
+
+  const [isOpen, setOpen] = useState(false);
+
+  const handleOpen = () => {
+    setOpen(true);
+    setSelectedDates([]);
+  };
+
+  const handleClose = () => {
+    setOpen(false);
+  };
+
+  const handleSubmit = async () => {
+    // Use YYYY-MM-DD format
+    const dateRange = selectedDates.map((date) =>
+      date.toISOString().substring(0, 10),
+    );
+
+    const data = await commonAPIClient.filterLogRecords(
+      undefined,
+      undefined,
+      undefined,
+      dateRange,
+      undefined,
+      undefined,
+      true, // return all data
+    );
+
+    setShowAlert(CSVConverter(data.logRecords) === "error");
+  };
+
+  useEffect(() => {
+    if (showAlert) {
+      setTimeout(() => {
+        setShowAlert(false);
+      }, 3000);
+    }
+  }, [showAlert]);
+
+  return (
+    <>
+      <IconButton
+        aria-label="Print CSV"
+        className="ghost-button"
+        icon={<Icon boxSize="32px" as={AiFillPrinter} />}
+        variant="ghost"
+        onClick={handleOpen}
+      />
+
+      <Box>
+        <Modal isOpen={isOpen} onClose={handleClose} size="xl">
+          <ModalOverlay />
+          <ModalContent>
+            <ModalHeader>Print CSV File</ModalHeader>
+            <ModalBody>
+              <FormControl>
+                <Flex gap="8px">
+                  <RangeDatepicker
+                    selectedDates={selectedDates}
```

Comment:
I think it's a little hard to discern that this is a date-input (maybe we wanna update the [designs too?](https://www.figma.com/file/4AStM5IGeHpMw94iU19AkG/Homepage?type=design&node-id=571-18362&t=CxnlYZB8mx2GbBPe-0)) 

We can add this prop to add some placeholder text so it's more indicative of what this component is for:
![Screen Shot 2023-06-04 at 12 46 04 AM](https://github.com/uwblueprint/supportive-housing/assets/53386475/ed107bd4-3dd3-4f99-b0b9-3f60661886c6)




```suggestion
                    selectedDates={selectedDates}
                    propsConfigs={{
                      inputProps: {
                        placeholder: 'DD/MM/YYYY -> DD/MM/YYYY'
                      },
                    }}
```

Reply:
there are no designs for this specific modal 😅 I'm good with the suggestion above though